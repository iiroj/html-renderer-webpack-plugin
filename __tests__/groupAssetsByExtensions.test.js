const { RawSource } = require('webpack-sources');
const { groupAssetsByExtensions } = require('../index');

const assets = {
  '5749aae7.js': new RawSource({
    _value: 'test'
  }),
  '2ff13846.js': new RawSource({
    _value: 'test'
  }),
  '5e6d94fc.css': new RawSource({
    _value: 'test'
  }),
  '1e77c6d6.css': new RawSource({
    _value: 'test'
  }),
  '7caf52ec.jpg': new RawSource({
    _value: 'test'
  }),
  '676c1b7d.png': new RawSource({
    _value: 'test'
  }),
  '85a81ee1': new RawSource({
    _value: 'test'
  })
};

describe('groupAssetsByExtensions', () => {
  it('should group assets by extensions', () => {
    const assetsByExtensions = groupAssetsByExtensions(assets);
    expect(assetsByExtensions).toMatchSnapshot();
  });
});
