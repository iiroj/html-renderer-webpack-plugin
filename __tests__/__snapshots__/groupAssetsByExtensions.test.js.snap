// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should group assets by extensions 1`] = `
Object {
  "": Array [
    "85a81ee1",
  ],
  "css": Array [
    "5e6d94fc.css",
    "1e77c6d6.css",
  ],
  "jpg": Array [
    "7caf52ec.jpg",
  ],
  "js": Array [
    "5749aae7.js",
    "2ff13846.js",
  ],
  "png": Array [
    "676c1b7d.png",
  ],
}
`;
